src/libraries/System.Private.Xml/src/System/Xml/Core/XmlTextReaderImpl.cs

```diff
@@ -3744,7 +3744,7 @@ private bool ParseXmlDeclaration(bool isTextDecl)
                 switch (_ps.chars[_ps.charPos])
                 {
                     case 'v':
-                        if (XmlConvert.StrEqual(_ps.chars, _ps.charPos, nameEndPos - _ps.charPos, "version") && xmlDeclState == 0)
+                        if (nameEndPos - _ps.charPos == "version".Length && _ps.StartsWith("version") && xmlDeclState == 0)
```

Comment:
Shouldn't this be
```suggestion
                        if (nameEndPos - _ps.charPos == "version".Length && _ps.AsSpan(_ps.charPos).StartsWith("version") && xmlDeclState == 0)
```
Same with all other changes?

Reply:
Also, could the particular switch statement be expressed as follows?
```C#
switch (_ps.chars.AsSpan(_ps.charPos, nameEndPos))
{
    case "version": /* ... */
    case "encoding": /* ... */
    case "standalone": /* ... */
}
```
This translates `ROS.SequenceEqual()` calls in IL.

Reply:
This sounds better. I will change it to 

```
switch (_ps.chars.AsSpan(_ps.charPos, nameEndPos - _ps.charPos))
{
    case "version":
        if (xmlDeclState == 0)
        {
            if (!isTextDecl)
            {
                attr = AddAttributeNoChecks("version", 1);
            }
            break;
        }
        goto default;
    case "encoding":
        if (xmlDeclState == 1 || (isTextDecl && xmlDeclState == 0))
        {
            if (!isTextDecl)
            {
                attr = AddAttributeNoChecks("encoding", 1);
            }
            xmlDeclState = 1;
            break;
        }
        goto default;
    case "standalone":
        if ((xmlDeclState == 1 || xmlDeclState == 2) && !isTextDecl)
        {
            attr = AddAttributeNoChecks("standalone", 1);
            xmlDeclState = 2;
            break;
        }
        goto default;
    default:
        Throw(isTextDecl ? SR.Xml_InvalidTextDecl : SR.Xml_InvalidXmlDecl);
        break;
}
```

Reply:
Done